Game/Source/GDKTestGyms/BenchmarkGymGameModeBase.cpp

```diff
@@ -157,6 +193,13 @@ void ABenchmarkGymGameModeBase::Tick(float DeltaSeconds)
 	TickClientFPSCheck(DeltaSeconds);
 	TickPlayersConnectedCheck(DeltaSeconds);
 	TickUXMetricCheck(DeltaSeconds);
+	TickActorCountCheck(DeltaSeconds);
+
+	if (HasAuthority() &&
+		PrintMetricsTimer.HasTimerGoneOff())
```

Comment:
This logic seems a little confusing - we're checking if it's "gone off" - but then just immediately resetting it. Why isn't the print logic tied up here?

Reply:
So the timer is just being reset so it is similar to a looping timer. Each "tick expectation" function has the chance to check if the timer went off and then print stats if it wants to. We could tie the print logic at this point but we would have to somehow get the relevant stats to print. These would all have to be member variables (some are already). That just didn't seem neat to me. Members for the purpose of passing information around just for the sake of logs.

Reply:
Okay - as there's a dependency on call order here could you add a brief comment detailing this usage - ie. give each tick function that ability to print using this timer, then reset.